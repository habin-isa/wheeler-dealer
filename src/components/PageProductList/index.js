import React, { useState, useCallback } from "react";
import { getProducts } from "../../services/products";

const PageProductList = () => {
  const [productData, setProductData] = useState([]);

  const loadProducts = useCallback(async () => {
    try {
      const response = await getProducts();
      setProductData(response);
    } catch (error) {
      console.log("error fetching products");
    } finally {
      console.log("ran");
    }
  }, [productData]);

  return (
    <div>
      FERRARI
      <button onClick={loadProducts}>LOAD</button>
    </div>
  );
};

export default PageProductList;