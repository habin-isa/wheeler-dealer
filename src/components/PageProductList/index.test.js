import React from 'react';
import PageProductList from '.';
import { render } from '@testing-library/react';

const props = {};

describe('PageProductList', () => {
    it('should render', () => {
        const { container } = render(<PageProductList {...props} />);
        expect(container.firstChild).toMatchSnapshot();
    });
});