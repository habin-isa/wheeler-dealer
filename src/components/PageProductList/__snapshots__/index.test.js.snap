// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageProductList should render 1`] = `
<div>
  <div
    class="sc-AxirZ dmCnmV"
  >
    <div>
       Results
    </div>
    <button>
      Hide sold items
    </button>
    <div>
      <img
        class="sc-AxiKw koNhYh"
        src="full-like.svg"
      />
       
      0
    </div>
  </div>
  <div
    class="sc-AxjAm kdAMH"
  />
</div>
`;
